include/odp/api/spec/schedule.h

```diff
@@ -45,6 +45,110 @@ extern "C" {
  * Maximum schedule group name length in chars including null char
  */
 
+/**
+ * Schedule configuration
+ * When schedule configuration is not called by the application, the scheduler
+ * behaves in flow-unaware mode and event flow hash values are ignored
+ */
+typedef struct odp_schedule_config_t {
+	/** Number of flows per queue to be supported
+	 * Scheduler enables flow aware mode when flow count is configured
+	 * greater than 1.
+	 *
+	 * Flows are lightweight entities and packets can be assigned to
+	 * specific flows by the application using odp_event_flow_hash_set()
+	 * before enqueuing the packet into the scheduler. This value is
+	 * ignored unless scheduler supports flow aware mode.
+	 *
+	 * Depening on the implementation this number might be rounded up to
+	 * nearest supported value (e.g power of 2). odp_schedule_config() will
+	 * update passed config with the value actually used. This number
+	 * should be less than maximum flow supported by the implementation.
+	 *
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t flow_count;
+
+	/** Maximum number of schedule queues to be supported
+	 * Application configures the maximum number of schedule queues to be
+	 * supported by the implementation.
+	 * @see odp_queue_capability_t
+	 */
+	uint32_t queue_count;
+
+	/** Maximum number of events required to be stored simultaneously in
+	 * schedule queue. This number should be less than 'max_queue_size'
+	 * supported by the implementation.
+	 * A value of 0 configures default queue size supported by the
+	 * implementation.
+	 */
+	uint32_t queue_size;
+} odp_schedule_config_t;
+
+/**
+ * Scheduler capabilities
+ */
+typedef struct odp_schedule_capability_t {
+	/** Maximum supported flows per queue
+	 * Specifies the maximum number of flows per queue supported by the
+	 * implementation.
+	 * A value of 0 indicates flow aware mode is not supported.
+	 */
+	uint32_t max_flow_count;
+
+	/** Maximum supported schedule queues
+	 * Specifies the maximum number of schedule queues supported by the
+	 * implementation.
+	 */
+	uint32_t max_queue_count;
+
+	/** Maximum number of events a schedule queue can store simultaneoulsy.
+	 * A value of 0 indicates the implementation does not restrict the
+	 * queue size
+	 */
+	uint32_t max_queue_size;
+
+	/** Maximum schedule group
+	 * Specifies the maximum number of schedule group supported by the
+	 * implementation.
+	 */
+	uint32_t max_schedule_group;
+} odp_schedule_capability_t;
```

Comment:
Information / naming conventions in this type should match those in queue_capa so that scheduler portion of  queue_capa can be deprecated.

Reply:
@psavol do you suggest to move all `queue_capa.sched` to scheduler capabilities?

Reply:
Actaully all scheduler capabilities:
.max_ordered_locks
.max_groups
.prios
.queue.max_num
.queue.max_size
.queue.lockfree.max_num
...